### PR TITLE
Feature/registration page styles

### LINKED ISSUE #1420: Create Registration page
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/1419

### Description

n/a

### LINKED ISSUE #1421: Registration Form layout and styles
### Link task to story

Story: https://github.com/Open-Study-College/osc/issues/1419

### Description

n/a

### PR DESCRIPTION
**Pull request by stevenbister**
### 🧹 Please check off the following before submitting:

- [x] Close issues
- [x] Request reviewers
- [x] Assign yourself
- [ ] Add labels (e.g. `build:osc-ecommerce`)
- [x] Add project
- [x] Set project status
- [x] Add milestone

---

## ☑ List all github issues that this PR will close:

- Closes #1420 
- Closes #1421 

## 📝 Description

Adds the registration form, styles and some initial logic for displaying certain fields/the different forms. This just styling and layout so won't include any functionality or live data right now.

## ⛳️ Current behaviour (updates)

N/a

## 🚀 New behaviour

- Adds registration form
- Adds complete registration form to set password etc.
- Adds layout and styles for registration page
- Redirect the registration page if user is logged in
- Switches form when the token url param is present
- Extends svg props onto `Icon` so we can pass classes, style tags etc.

## 💣 Is this a breaking change (Yes/No): No

> Please describe any breaking changes.

## 🧾 Final Checklist

- [ ] I've considered possible cross-platform impacts (Mac, Linux, Windows)
- [x] I've considered possible cross-device impacts (iOS, Android)
- [x] I've considered possible cross-browser impacts (Chrome, Safari, Edge, Firefox)
- [x] I've added tests to cover my changes (Unit, E2E, Integration, Contract, Functional, Acceptance etc)
- [ ] I've added or updated any documentation.

## 📝 Additional Information

Like the other AH prs if this is your first one you may need some new env vars, let me know and I can provide you them.

To run the app you'll need to run the api first `npm run dev:api` followed by `npm run dev:osc-academic-hub`

To test the complete registration form you'll need to add the token url param for example (`?token=123`)

### Summary by CodeRabbit

- New Feature: Introduced `RegistrationForm` and `CompleteRegistrationForm` components for user registration and completion of the registration process.
- New Feature: Added `SenInput` component to handle Special Educational Needs (SEN) input during registration.
- New Feature: Implemented `IconText` component for displaying icons with labels.
- Refactor: Updated `AuthCardPanel` component to support customization of width and background opacity.
- Style: Enhanced visual appearance and responsiveness of the authentication card component with new CSS classes and variables.
- Test: Added test cases for `LoginForm`, `RegistrationForm`, and `CompleteRegistrationForm` components.
- Chore: Updated various modules, stylesheets, and functions across multiple files for improved functionality and code structure.

### CODE REVIEW

File: packages/osc-academic-hub/app/components/Registration/RegistrationForm.tsx [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+                    <TextInput
+                        id="email"
+                        name="email"
+                        label="Email"
+                        type="email"
+                        variants={['secondary']}
+
+                        // errors={fieldErrors?.email}
+                        // setErrors={setFieldErrors}
+                        // schema={auth.pick({ email: true })}
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-academic-hub/app/components/Registration/RegistrationForm.tsx:50]: These are commented out intentionally and will be added in a later pr

**Review by hi-im-si — CHANGES_REQUESTED** [review]
As discussed everything is looking good but i think keeping some consistency with the layout/structure/styles between the login and register pages would be beneficial i.e.
the use of `c-auth-card` to be used on /register.
This will allow us to rely less on utilities and promote reusability. 
I guess what i am trying to allude to is that it would be a shame to create an scss component only to ever be used on one page.

**Comment by stevenbister** [comment]
> As discussed everything is looking good but i think keeping some consistency with the layout/structure/styles between the login and register pages would be beneficial i.e.
> the use of `c-auth-card` to be used on /register.
> This will allow us to rely less on utilities and promote reusability.
> I guess what i am trying to allude to is that it would be a shame to create an scss component only to ever be used on one page.

Yeah that's fair, I've updated the auth card to have a couple of props/modifiers which makes it more flexible and added it to the registration layout

File: packages/osc-academic-hub/app/components/Auth/auth-card.scss [hunk 2]
```diff
@@ -3,7 +3,10 @@
 
 @layer components {
     .c-auth-card {
-        $section-height-limit: 600px;
+        $auth-card-padding-inline-s: $space-3xl;
+        $auth-card-padding-inline-l: $space-4xl;
+        $auth-card-padding-block-s: $space-2xl;
+        $auth-card-padding-block-l: $space-4xl;
 
         padding-block: $space-l;
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-academic-hub/app/components/Auth/auth-card.scss:12]: We want to keep these variables generic as to not tie them into specifically being for the first or last child, they get used in other places such as `only-child`

File: packages/osc-academic-hub/app/routes/register.tsx [hunk 3]
```diff
@@ -1,13 +1,164 @@
...
+    const getTokenFromUrl = url.searchParams.get('token')?.trim();
+
+    if (getTokenFromUrl) {
+        return json<LoaderData>({
+            token: getTokenFromUrl,
+        });
+    }
+
+    return json<LoaderData>({});
+};
```
[the remark this answers is not in the document]
Comment (stevenbister) [reply packages/osc-academic-hub/app/routes/register.tsx:46]: Yeah, that's intentionally here. This is part one of a two part piece of work and will be addressed in a later PR

**Review by hi-im-si — APPROVED** [review]
Looking good!

**Pull request merged**